src/network/address.rs

```diff
@@ -110,12 +110,186 @@ impl fmt::Debug for Address {
     }
 }
 
+/// Address received from BIP155 addrv2 message
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+pub struct AddrV2Address {
```

Comment:
W.r.t naming, maybe `AddrV2Msg` for the message and just `AddrV2` thus avoiding the reference to IP.